### ISSUE TITLE
Alt text still uses image title even when displaying "image not available" 

### ISSUE THREAD

**Issue by sarayourfriend** [WordPress/openverse#4665]
## Description

The default alt text of the image title is still used even if "image not available" is shown (thumbnail/upstream image request error).

![image](https://github.com/user-attachments/assets/37f9baad-eb35-4014-8cb0-3aa06d96f8d8)

As far as I can tell, there is zero context available on this page to inform users relying exclusively on screen readers (or related tech) 

## Reproduction

1.  Visit https://openverse.org/image/2dc93cee-34d6-4b98-a76e-92ebc8d4ae73 (this image is unavailable but still present on Openverse at time of issue creation, if the image is gone or is available again, you may need to find another example to work with)
2.  Check the alt text of the image and see that it is still the image title.

## Additional context

I am making the image request from the UK, without a VPN, from a temporary location. It may be that the image is unavailable from the UK specifically, and is thus not filtered as dead when the API checks it _or_ this was available at one point and the API is used a cached check leading to it being included despite not being available.

You can find the specific result I linked above (at time of issue creation) on page 3 of [a search for "phytoplankton bloom"](https://openverse.org/search/?q=phytoplankton%20bloom).

This issue just need to address the fact that the alt text (and the page generally, but really specifically the alt text) does not indicate to screen-reader users that the image is not available. I've marked it a high priority a11y fix as such, because it severely impacts the usability and reliability of the Openverse website for someone in this situation and would be very frustrating to experience first hand (especially if a big group of results were unavailable, as is the case with the related images of the one linked above).

**Comment by sarayourfriend** [comment]
Worth asking whether the frontend should pop these images out of the search results when the thumbnail request fails: however, how to do that without causing a big shift in the search layout will be tricky. Solving the immediate issue of communicating unavailability of the image in text (some way screen-readers can understand) would solve the accessibility issue now. Another issue should be opened to address how these are handled, or it can be folded into the long-existing #510.

**Comment by zackkrida** [comment]
@sarayourfriend thinking about this more, I wonder if it would make sense to make these elements non-navigable by keyboard, i.e., give them a negative tabindex. That, combined with an approach to replace the image that failed to load: 

Switch the rendered fallback image, which currently contains inaccessible and ugly (depending on the screen size /zoom level/etc.) raster text, to an HTML element containing text, when the image fails to load. By not rendering an image we remove the need to set alt text, and screen readers will read out the actual text of the placeholder element instead.

**Comment by zackkrida** [comment]
Other members of @WordPress/openverse-frontend, could you share some thoughts here?

**Comment by obulat** [comment]
I think there are 2 possible reasons for a thumbnail to not be available:
1. the image does not exist on the source, even though the direct URL does not return an error
2. the image is of an unsupported format (here, it's `.ogv`, which is actually a video file)

If we make such images non-navigable, the users will not be able to open a [single result page](https://openverse.org/image/2dc93cee-34d6-4b98-a76e-92ebc8d4ae73?q=phytoplankton+bloom) for it. This means that it will be impossible go to the source of the image. This can be useful for the second kind of images, when the source might [show](https://commons.wikimedia.org/w/index.php?curid=59122955) the image correctly.

Maybe we should leave the link navigable, but replace the figure element with the text. This is a very rough draft:
<img width="200" alt="Screenshot 2024-08-05 at 6 49 33 PM" src="https://github.com/user-attachments/assets/9ed146c3-3f29-4b4b-b9fb-83444111b15f">

**Comment by sarayourfriend** [comment]
> This can be useful for the second kind of images, when the source might [show](https://commons.wikimedia.org/w/index.php?curid=59122955) the image correctly.

This is such a narrow use case (the vast majority of our image data are true images that are available and we can see that they are available), I don't know if we can easily communicate that in the UI without a _lot_ of verbiage.

It might mistake a problem with our data or service (e.g.: we incorrectly include video files in our image index, we have poor support for diverse image file types [if that is the case]) with needing to make anything possible in the frontend UI.

The frontend image search is for finding images. If there's plausible doubt that an image isn't available, I think the best thing is to prevent the frontend from giving a confusing experience to a user (what this discussion has turned into) followed by understanding the cases in which this happens and effectively tracking them so we can implement fixes (either to our data or our systems).

It's kind of like: the frontend should display the good data we have. It doesn't need to spend time trying to make the "junk" data usable, especially because we control that data to begin with. If there's junk data that the frontend can't fit into "an easy to understand and use image search", then we should develop ways of finding out those instances so we can fix them upstream. The frontend isn't for "data spelunking" the Openverse API, it's for finding usable images, and thumbnail previews are a core part of how that works.

Said another way: the frontend could be thought of as the happy path, with temporary fixes for small data issues if needed, but ideally no long-term need to manipulate or interpret the data from the API in complex ways. That has the advantage of putting our development of the frontend in the mindset of any other API user who just has to use the data as is, and would probably end up frustrated with the quality of it if they ran into this.

With that in mind, I think the best solution overall is to exclude these from search somehow, like Zack suggested. But to solve the accessibility issue, which is the only thing this specific issue is actually about, despite the discussion diverging a bit, is theoretically much simpler.

We should probably have at least two more issues though:

1. Some way of tracking (and actually following up on) cases where the frontend wasn't able to load an image: this might be better placed in the thumbnails route, as a secondary place where we naturally do link liveness checking, beyond the check-dead links routine. The "flaky thumbnails" for example, might be a good signal that we should consider that result dead, keeping in mind #428 can incorporate that data in diverse ways, including more robust checks of upstream liveness that the API doesn't have time for at search.
2. Figure out how to exclude these from search on the frontend to prevent temporarily bad data (temporary because we're doing the first issue) from ruining search or ruining it.

What do you think, @obulat? Does this make sense, do you agree or disagree with my approach to thinking about this?

**Comment by fcoveram** [comment]
> The frontend image search is for finding images. If there's plausible doubt that an image isn't available, I think the best thing is to prevent the frontend from giving a confusing experience to a user 

+1 to this. Showing a result where the image can't be seen within image results contradicts the core purpose of searching. Avoiding dev hours in making junk data usable is a great approach to these kind of use cases.

Assuming these items will be removed from the results, what would happen with @obulat's example where the image is visible in the source view? I mean [this one](https://openverse.org/image/collection?source=wikimedia&creator=Lehahn+Y,+Koren+I,+Sharoni+S,+d'Ovidio+F,+Vardi+A,+Boss+E). Would it be removed as well?

On the other hand, it seems like a good opportunity to address #510

**Comment by sarayourfriend** [comment]
> Assuming these items will be removed from the results, what would happen with @obulat's example where the image is visible in the source view? I mean [this one](https://openverse.org/image/collection?source=wikimedia&creator=Lehahn+Y,+Koren+I,+Sharoni+S,+d%27Ovidio+F,+Vardi+A,+Boss+E). Would it be removed as well?

For that _specific_ case, It's only visible there due to us using a new method for Wikimedia thumbnail generation that incidentally supports thumbnails for videos (that wasn't the intention of the change, just a side effect).

We'll remove those videos with #4717 so they won't be there any more. No junk data. That source page will be empty after that.

Speaking to the general case though, if an image is not available, it won't be on the source page either (we use the same methods for both source and search to display result thumbnails). Both pages will receive the same treatment (the result removed from the list and not shown on the page).

It seems that this would potentially supersede #510 by just excluding unavailable results rather than falling back. On the other hand, we can't generate thumbnails for non-Wikimedia TIFFs and SVGs; but it would be great if we could find an alternative thumbnail generation method on the API side to handle those if Photon won't. I don't really know what that would be where it wouldn't potentially cost us a rather large sum of money in processing and storage, or who or when we would implement such a thing. I guess we do need _some_ fallback behaviour, and I don't like the idea of streaming an uncompressed TIFF scan of a huge archival document as the "fallback". It would be helpful to know how many sources really have SVGs and TIFFs where we can't get a thumbnail from the source. Maybe this is a non-problem. Flickr doesn't have SVGs or TIFFs as far as I can tell from what's available in public documentation. It would be good to know that rather than being worried we aren't supporting some big important subset of our data.
